platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/schemaBuilder/ConfigureLambdaHandler.kt

```diff
@@ -115,8 +116,10 @@ fun treatInterfaceAsConfigureLambda(functionalInterface: KClass<*>): ConfigureLa
     }
 
     init {
-        check(functionalInterface.java.isInterface)
-        check(typeParameters.size <= 1) { "generic types with more than one type parameter are not supported" }
+        check(functionalInterface.java.isInterface) // TODO: can this happen due to user error?
```

Comment:
> // TODO: can this happen due to user error?

It is a cross-subsystem invariant, as the `functionalInterface` is a part of the DCL setup outside of schema building. But Gradle users cannot really affect this currently.